### PR TITLE
RUMM-310 navigation

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Create a RUM integration for AndroidX Navigation framework

### Review checklist (to be filled by reviewers)

- [X] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [X] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [X] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: buildSrc/src/main/kotlin/com/datadog/gradle/Dependencies.kt [hunk 1]
```diff
@@ -162,6 +173,7 @@ object Dependencies {
         const val KotlinAndroid = "org.jetbrains.kotlin.android"
         const val KotlinAndroidExtension = "org.jetbrains.kotlin.android.extensions"
         const val Bintray = "com.jfrog.bintray"
+        const val Unmock = "de.mobilej.unmock"
```
Comment (mariusc83) [inline buildSrc/src/main/kotlin/com/datadog/gradle/Dependencies.kt]: Oh this is nice ;)

File: dd-sdk-android-navigation/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import com.datadog.gradle.config.junitConfig
+import com.datadog.gradle.config.kotlinConfig
+import com.datadog.gradle.config.ktLintConfig
+import com.datadog.gradle.config.publishingConfig
+import com.datadog.gradle.implementation
+import com.datadog.gradle.testImplementation
+
+plugins {
+    id("com.android.library")
+    id("androidx.benchmark")
```
Comment (mariusc83) [inline dd-sdk-android-navigation/build.gradle.kts]: Why do you need this ?

File: dd-sdk-android-navigation/src/main/kotlin/com/datadog/android/androidx/navigation/NavigationViewTrackingStrategy.kt [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        }
+    }
+
+    private fun Bundle?.asRumAttributes(): Map<String, Any?> {
+        if (this == null) return emptyMap()
+
+        val attributes = mutableMapOf<String, Any?>()
+
+        keySet().forEach {
+            attributes["$ARGUMENT_TAG.$it"] = get(it)
```
Comment (mariusc83) [inline dd-sdk-android-navigation/src/main/kotlin/com/datadog/android/androidx/navigation/NavigationViewTrackingStrategy.kt]: hmmm....I am not sure this is GDPR compliant. What if we have sensible information in the arguments ?

File: buildSrc/src/main/kotlin/com/datadog/gradle/Dependencies.kt [hunk 4]
```diff
@@ -37,6 +41,8 @@ object Dependencies {
         const val DependencyVersion = "0.27.0"
         const val Dokka = "0.10.0"
         const val Bintray = "1.8.4"
+        const val Unmock = "0.7.5"
+        const val Robolectric = "4.4_r1-robolectric-r2" // Use lowest API
```
Comment (mariusc83) [inline buildSrc/src/main/kotlin/com/datadog/gradle/Dependencies.kt]: Why do we need Robolectric ? I guess your intention was to use this to test your new strategy but then you found the "unmock" solution.

Comment (xgouchet) [reply buildSrc/src/main/kotlin/com/datadog/gradle/Dependencies.kt]: We don't use the Robolectric runner, just the jar that implements the Android framework. 
Unmock let's you load only the required class from the Robolectric jar, which is used to unit test (among other things, it's needed to test against the Bundle class)

File: dd-sdk-android-navigation/build.gradle.kts (hunk 2 above)
Comment (xgouchet) [reply dd-sdk-android-navigation/build.gradle.kts]: hmm not needed indeed.

File: dd-sdk-android-navigation/src/main/kotlin/com/datadog/android/androidx/navigation/NavigationViewTrackingStrategy.kt (hunk 3 above)
Comment (xgouchet) [reply dd-sdk-android-navigation/src/main/kotlin/com/datadog/android/androidx/navigation/NavigationViewTrackingStrategy.kt]: Good point. 
I'll add an argument to the class to let the developper enable this feature, which I think is really usefull. 
Also most of the time the data passed in arguments are item ids, and not necessarily "sensitive" data .

**Pull request merged**